cmd/sonobuoy/app/operations/logs.go

```diff
@@ -17,11 +17,66 @@ limitations under the License.
 package operations
 
 import (
-	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/heptio/sonobuoy/pkg/config"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	podLogSeparator = strings.Repeat("-", 79)
 )
 
-// Logs gathers the logs for the containers in the sonobuoy namespace and prints them out in a structured format
-func GetLogs( /*opts?*/ ) error {
-	// This should be straight forwards
-	return errors.New("not implemented")
+// Logs gathers the logs for the containers in the sonobuoy namespace and prints them
+
+// GetLogs streams logs from the sonobuoy pod by default to stdout.
+func GetLogs(cfg *config.Config, follow bool) error {
+	client, err := config.OutOfClusterClient()
+	if err != nil {
+		return errors.Wrap(err, "could not load kubernetes client")
+	}
+
+	// Follow the main logs if -f is provided
+	if follow {
+		return streamLogs(client, cfg.PluginNamespace, config.SonobuoyPod, &v1.PodLogOptions{Follow: true})
+	}
+
+	// Otherwise print every log we can
+	pods, err := client.CoreV1().Pods(cfg.PluginNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "could not list pods")
+	}
+	for _, pod := range pods.Items {
+		for _, container := range pod.Spec.Containers {
+			logrus.WithFields(logrus.Fields{"pod": pod.Name, "container": container.Name}).Info("Printing container logs")
+			err := streamLogs(client, cfg.PluginNamespace, pod.Name, &v1.PodLogOptions{
+				Container: container.Name,
+			})
+			if err != nil {
+				return errors.Wrap(err, "failed to stream logs")
+			}
+			fmt.Println(podLogSeparator)
```

Comment:
:/ not happy about this. Think it's alright?

Reply:
I was also thinking of disabling this line with -q but maybe we can implement that later if need be.

Reply:
I'm ok with it.  Eventually you may want a utility f(n) like prettyPrint for a container with an optional separator arg.